packages/blockchain/src/types.ts

```diff
@@ -34,9 +35,45 @@ export interface BlockchainInterface {
    *
    * @param name - Name of the state root head
    * @param onBlock - Function called on each block with params (block: Block,
+   * @param maxBlocks - optional maximum number of blocks to iterate through
    * reorg: boolean)
    */
-  iterator(name: string, onBlock: OnBlock): Promise<number>
+  iterator(name: string, onBlock: OnBlock, maxBlocks?: number): Promise<number>
+
+  /**
+   * Returns a copy of the blockchain
+   */
+  copy(): BlockchainInterface
+
+  /**
+   * Returns the specified iterator head.
+   *
+   * @param name - Optional name of the iterator head (default: 'vm')
+   */
+
+  /**
+   * Validates a block header, throwing if invalid. It is being validated against the reported `parentHash`.
+   * @param header - header to be validated
+   * @param height - If this is an uncle header, this is the height of the block that is including it
+   */
+  validateHeader(header: BlockHeader, height?: bigint): Promise<void>
+  getIteratorHead?(name?: string): Promise<Block>
```

Comment:
Missing docs

Reply:
Docs are a bit more above, have added to the tiny-fixes issue (that was the trigger 😋)